app/models/proposal.rb

```diff
@@ -118,6 +121,14 @@ def votable_by?(user)
     user && user.level_two_or_three_verified?
   end
 
+  def valid_video_url?
+    return if video_url.blank?
```

Comment:
I think the constant holding the valid video domains and the body of this function should be shared on a common helper for the near future (other models are for sure getting video urls this next week :) ). Like at https://github.com/consul/consul/blob/master/app/helpers/embed_videos_helper.rb. And this `valid_video_url?` could just be using it.